src/main/java/org/util/string/StringUtil.java

```diff
@@ -31,268 +34,244 @@
      */
     public StringUtil append(StringBuffer data) {
-        return new StringUtil(new StringBuilder(builder).append(data));
+        StringUtil other = new StringUtil();
+        other.builder.append(builder);
+        other.builder.append(data);
+        return other;
     }
 
     /**
-     * void append String data
-     * this method append String data to field StringBuilder builder
+     * Appends data to the end. Does not change state of this instance.
      *
-     * @param data
-     * @returns void
+     * @param data - value to append
+     * @return another object with modified state
      */
     public StringUtil append(String data) {
-        // wrong - inefficient
-        return new StringUtil(builder.toString().concat(data));
+        StringUtil other = new StringUtil();
+        other.builder.append(builder);
+        other.builder.append(data);
+        return other;
     }
 
     /**
-     * void append StringBui
-     * lder data
-     * this method append StringBuilder data to field StringBuilder builder
+     * Appends data to the end. Does not change state of this instance.
      *
-     * @param data
-     * @returns void
+     * @param data - value to append
+     * @return another object with modified state
      */
     public StringUtil append(StringBuilder data) {
-        // wrong - inefficient
-        //this.builder.append(data);
-//        return new StringUtil(this.builder.toString() + data);
-        return new StringUtil(builder.toString().concat(String.valueOf(data)));
+        StringUtil other = new StringUtil();
+        other.builder.append(builder);
+        other.builder.append(data);
+        return other;
     }
 
     /**
-     * void append  data
-     * this method append StringBuilder data to field StringBuilder builder
+     * Appends data to the end. Does not change state of this instance.
      *
-     * @param data
-     * @returns void
+     * @param data - value to append
+     * @return another object with modified state
      */
     public StringUtil append(char[] data) {
-        // wrong - inefficient
-        return new StringUtil(builder.toString().concat(String.valueOf(data)));
+        StringUtil other = new StringUtil();
+        other.builder.append(builder);
+        other.builder.append(data);
+        return other;
     }
 
     /**
-     * void append data from 'byte[] data'  with 'Charset charset' over call of constructor with
-     * parameters types (byte[] data, Charset charset)
+     * Appends data to the end. Does not change state of this instance.
      *
-     * @param data
-     * @returns void
+     * @param data    - value to append
+     * @param charset
+     * @return another object with modified state
      */
     public StringUtil append(byte[] data, Charset charset) {
-        return new StringUtil(builder.toString() + new StringUtil(data, charset).toString());
+        StringUtil other = new StringUtil();
```

Comment:
1

Reply:
"".getBytes() // add charset to method call
create charset: Charset.forName("UTF-8")